homeassistant/components/here_weather/config_flow.py

```diff
@@ -0,0 +1,84 @@
+"""Config flow for here_weather integration."""
+from __future__ import annotations
+
+import herepy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+
+from .const import DEFAULT_MODE, DOMAIN
+
+
+async def async_validate_user_input(hass: HomeAssistant, user_input: dict) -> None:
+    """Validate the user_input containing coordinates."""
+    here_client = herepy.DestinationWeatherApi(user_input[CONF_API_KEY])
+    await hass.async_add_executor_job(
+        here_client.weather_for_coordinates,
+        user_input[CONF_LATITUDE],
+        user_input[CONF_LONGITUDE],
+        herepy.WeatherProductType[DEFAULT_MODE],
+    )
+
+
+class HereWeatherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for here_weather."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            try:
+                await self.async_set_unique_id(_unique_id(user_input))
```

Comment:
Please only wrap the line that can raise in the try... except block. Use an `else:` block if needed.

Reply:
Done in https://github.com/home-assistant/core/pull/28910/commits/ac010a6f8648ab45cc10e0c8ace2dd599323ffcc